spec/classes/candlepin_spec.rb

```diff
@@ -416,6 +416,17 @@
         it { is_expected.to contain_exec('notification').that_notifies('Service[tomcat]') }
         it { is_expected.to contain_exec('dependency').that_requires('Service[tomcat]') }
       end
+
+      describe 'with facts match regex' do
+        let(:params) { { facts_match_regex: 'test_match_regex' } }
+
+        it { is_expected.to compile.with_all_deps }
+
+        it do
+          is_expected.to contain_file('/etc/candlepin/candlepin.conf').
```

Comment:
```suggestion
          is_expected.to contain_concat__fragment('General Config').
```

Take a look at how it's done here -- https://github.com/theforeman/puppet-candlepin/pull/262/files#diff-b55cd2d4fee4ad88a9e21316ff2015c5579e8ec9bef7bc0fe5253262905fa702R36